process/smartContract/process.go

```diff
@@ -649,6 +735,10 @@ func (sc *scProcessor) processSCOutputAccounts(outputAccounts []*vmcommon.Output
 		// update the values according to SC output
 		updatedBalance := big.NewInt(0)
 		updatedBalance = updatedBalance.Add(stAcc.Balance, outAcc.BalanceDelta)
+		if updatedBalance.Cmp(big.NewInt(0)) < 0 {
```

Comment:
Don't we have a problem in the vm if the balanceDelta is negative and smaller than the balance? Why would the vm process something like this?

Reply:
balanceDelta will be negative for some accounts, in the VM the APIs are protecting from bad behaviour. In case of lottery, when the SC announces the winner, the balanceDelta of the SC will be negative, and the others will be positive.

Reply:
ok